presto-main/src/main/java/com/facebook/presto/transaction/InMemoryTransactionManager.java

```diff
@@ -417,6 +418,14 @@ public void checkOpenTransaction()
                 if (catalog.isPresent()) {
                     registerCatalog(catalog.get());
                 }
+
+                if (catalogManager.hasFederations(catalogName)) {
+                    Set<Optional<Catalog>> federatedCatalogs = catalogManager.getFederatedCatalogs(catalogName);
+                    if (!federatedCatalogs.stream().allMatch(Optional::isPresent)) {
+                        throw new IllegalStateException("Not all federated catalogs exist");
+                    }
+                    federatedCatalogs.forEach(federatedCatalog -> registerCatalog(federatedCatalog.get()));
```

Comment:
This isn't necessary. `CatalogManager` is injected. All catalogs should be visible as long as they are installed.

Reply:
So the purpose of this is to expose a set of "federated" catalogs. Catalogs are indeed visible but the visibility is not the issue - the issue is that we need a way to know which set of catalogs should be installed for a given connector. I.e. when a hive transaction starts, it does not care for scuba transactions at all. Without this we would have to arbitrarily allow access to any catalogs in catalog manager which may be unsafe due to any ACL logic.

Might be a better way to do this though.